### PR TITLE
feat:(TON-2667) adapter read functions

### PR DESCRIPTION
**Pull request by sjDu**
## Summary

- Add part1 functions

## Ticket(s)
https://tonfura.atlassian.net/browse/TON-2667

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor (no functionality changes)

## Reference Document(s)

https://www.notion.so/frigatebird/TonClient4-Adapter-Feasibility-Study-TON-2481-d41312d06aac48ab94d446dc75d2a756?pvs=4#36beba3fc64149c5bb00035ee57cbbb5

## Checklist

- [ ] Remove unnecessary console or comment
- [x] Add unit tests

### CODE REVIEW

File: packages/adapter/src/tonclient4/converters/convertGetAccountTransactions.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import { GetTransactionsResponse, AccountTransactions, Message, CommonMessageInfo } from '../types'
+import { convertHexShardToSignedNumberStr, convertRawAddressToDecimalBigInt, decodeBase64ToDecimal, decodeBase64ToUnit8Array } from '../utils'
+import { Address, ExternalAddress } from '@ton/core';
+
+const getMessagaeType = (message: {
```
Comment (nashPOP) [inline packages/adapter/src/tonclient4/converters/convertGetAccountTransactions.ts]: TYPO should be getMessageType

File: packages/adapter/src/tonclient4/converters/convertGetAccountTransactions.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      type = 'external-in';
+    }
+  }
+
+  return type;
+}
+
+
+function getMessageInfo(message: Message): CommonMessageInfo {
+  const type = getMessagaeType(message);
```
Comment (nashPOP) [inline packages/adapter/src/tonclient4/converters/convertGetAccountTransactions.ts]: TYPO should be getMessageType

File: packages/adapter/src/tonclient4/converters/convertGetAccountTransactions.ts (hunk 1 above)
Comment (sjDu) [reply packages/adapter/src/tonclient4/converters/convertGetAccountTransactions.ts]: Good catch! Thanks Allen

File: packages/adapter/src/tonclient4/converters/convertGetBlock.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  return [
+    {
+      method: "getBlockTransactions",
+      params: {
+        workchain: -1,
+        shard: "8000000000000000",
+        seqno,
+      } 
+    },
+    ...data.result.shards.map(it => {
```
Comment (nashPOP) [inline packages/adapter/src/tonclient4/converters/convertGetBlock.ts]: `
  const shardInputs = data.result.shards.map(({ workchain, shard, seqno }) => ({
    method: "getBlockTransactions",
    params: { workchain, shard, seqno }
  }));
`  
  is more clean (optional)

**Review by nashPOP — APPROVED** [review]

File: packages/adapter/src/tonclient4/converters/convertGetBlock.ts (hunk 3 above)
Comment (sjDu) [reply packages/adapter/src/tonclient4/converters/convertGetBlock.ts]: Sure

**Pull request merged**